apache2/waf_logging/waf_log_util.cc

```diff
@@ -0,0 +1,104 @@
+#include "waf_log_util_internal.h"
+#include "waf_log_util_external.h"
+
+// This function fills in a Person message based on user input.
+void set_waf_format(waf_format::Waf_Format* waf_format, char* clientIP, char* clientPort, char* requestUri, char* ruleSetType, char* ruleSetVersion, char* ruleId, char* messages, int action, int site, char* details_messages, char* details_data, char* details_file, char* details_line, char* hostname) {
```

Comment:
Do we need to do some failure check in this function?

Reply:
put in the try catch